src/plugins/intel_gpu/src/kernel_selector/cl_kernels/search_sorted_ref.cl

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/fetch_data.cl"
+
+#if RIGHT_MODE == 0
+#define CMP <=
+#else
+#define CMP <
+#endif
+
+OUTPUT_TYPE binary_search_thread(const INPUT0_TYPE search_val,
```

Comment:
Please use FUNC() and FUNC_CALL() wrappers for functions to guarantee unique function names

Reply:
Fixed.